logic/hosts.go

```diff
@@ -313,3 +313,53 @@ func GetRelatedHosts(hostID string) []models.Host {
 	}
 	return relatedHosts
 }
+
+// CheckHostPort checks host endpoints to ensures that hosts on the same server
+// with the same endpoint have different listen ports
+// in the case of 64535 hosts or more with same endpoint, ports will not be changed
+func CheckHostPorts(h *models.Host) {
+	listenPort := h.ListenPort
+	proxyPort := h.ProxyListenPort
+	count := 0
+	reset := false
+	hosts, err := GetAllHosts()
+	if err != nil {
+		return
+	}
+	for _, host := range hosts {
+		if host.EndpointIP.Equal(h.EndpointIP) {
+			if host.ListenPort == h.ListenPort || host.ProxyListenPort == h.ProxyListenPort {
```

Comment:
we need to compare if the listenport == listenport || listenport == proxylistenport && proxyport == proxyport || proxyport == listenport, as if you had 100 machines on same NAT they could start to overlap on default ports